src/corecel/io/Logger.hh

```diff
@@ -44,17 +49,24 @@
     ::celeritas::world_logger()(CELER_CODE_PROVENANCE, \
                                 ::celeritas::LogLevel::LEVEL)
 
-//---------------------------------------------------------------------------//
 /*!
  * \def CELER_LOG_LOCAL
  *
  * Like \c CELER_LOG but for code paths that may only happen on a single
- * process. Use sparingly.
+ * process or thread. Use sparingly.
  */
 #define CELER_LOG_LOCAL(LEVEL)                        \
     ::celeritas::self_logger()(CELER_CODE_PROVENANCE, \
                                ::celeritas::LogLevel::LEVEL)
 
+// Allow CELER_LOG to be present (but ignored) in device code
+#if CELER_DEVICE_COMPILE
```

Comment:
This is following a different pattern than the other uses of `CELER_DEVICE_COMPILE` where it is more often, in place, `#if CELER_DEVICE_COMPILE ... do something ... #else .... do something else ... #endif`.  You may want to augment the commit log to give a slight justification of why the code in the PR is the better choice here.  (i.e. the alternative would be the self_logger thingy to have embedded the `nop` `device` functions)

Reply:
Yeah I wrote it like this because the use case here is "well, in this weird case let's suppress the logger so that we can at least compile".

> the self_logger thingy to have embedded the nop device functions

Not sure what that means; but if you're talking about using the self/world functions, they would have to be host/device and return a mutable reference, I don't think that's the right approach.

Reply:
I was unprecise.  What I meant is that the pattern we have elsewhere is to have the alternative inside `MessageLogger` itself and to have for example in `LoggerMessage.hh`
```
#if CELER_DEVICE_COMPILE
    //! Do not set any state
    CELER_FORCEINLINE_FUNCTION void setstate(std::ostream::iostate) {}
#else
    // Update the stream state
    inline void setstate(std::ostream::iostate state);
#endif
```
Both solution `works` but since the code in this PR deviates a bit from the norm, I think we should sprinkle the commit log with a vague explanation for that deviation :)

Reply:
We don't have that pattern elsewhere, I'm not sure it works for HIP, and it definitely won't work via the host-only global `Logger& world_logger()` function, for example...

Reply:
> We don't have that pattern elsewhere,

Humm ... right ... instead we have the `#if CELER_DEVICE_COMPILE` inside the functions body.  So the difference here could be "Replace whole class" instead of "modifying all functions" to adapt to device compilation.

>  and it definitely won't work via the host-only global Logger& world_logger() function, for example...

Fair enough (i.e. this is an additional note for the commit log :) )